apps/tx-builder/src/hooks/useTransactions/index.tsx

```diff
@@ -14,6 +14,19 @@ export default function useTransactions() {
     [transactions],
   );
 
+  const handleReplaceTransaction = useCallback(
```

Comment:
I think that we can reduce a lot the complexity of this function sending also de index of the transaction:

```
  const handleReplaceTransaction = useCallback(
    (newTransaction: ProposedTransaction, index: number) => {
      transactions[index] = newTransaction;
      setTransactions(transactions);
    },
    [transactions],
  );
```